.github/workflows/update-submodules.yml

```diff
@@ -17,21 +24,35 @@ jobs:
         uses: actions/checkout@v2
         with:
           submodules: true
-          persist-credentials: false
+          persist-credentials: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
 
-      - uses: actions/setup-go@v2
-        with:
-          go-version: 1.17
+      - name: Create and switch branch
+        env:
+          BRANCH: ${{ github.event.inputs.branch }}
+        run: |
+          if [[ $BRANCH == master ]]; then BRANCH=main; fi
+          git switch "$BRANCH" || git switch -c "$BRANCH"
 
-      - name: Update Git Sumbodule
+      - name: Update Submodule branch
+        env:
+          SUBMODULE: ${{ github.event.inputs.repo }}
+          BRANCH: ${{ github.event.inputs.branch }}
         run: |
-          git pull --recurse-submodules
-          git submodule update --remote --recursive
+          git config --file=.gitmodules submodule."$SUBMODULE".branch "$BRANCH"
+          git submodule update --init --recursive --remote
 
       - name: Commit update
+        env:
+          CURRENT_REPO: ${{ github.repository }}
+          TOKEN: ${{ secrets.COMMANDER_DATA_TOKEN }}
```

Comment:
why are there two different tokens? can they be combined to one?

Reply:
short answer is to stick to the minimal permissions necessary for a particular action.
Ideally we would only use `GITHUB_TOKEN` as it is new for every workflow run, however it doesn't allow to write into a protected branch, so the other commander token allows to do that

Reply:
yea it wasn't. Added the build trigger in the current PR https://github.com/temporalio/docker-builds/pull/23/commits/bc9706130e7761bb844dd93253862f3d7a2de5ee